crates/bitwarden-crypto/src/keys/key_ref.rs

```diff
@@ -0,0 +1,89 @@
+use crate::{AsymmetricCryptoKey, SymmetricCryptoKey};
+
+// Hide the `KeyRef` trait from the public API, to avoid confusion
+// the trait itself needs to be public to reference it in the macro, so wrap it in a hidden module
+#[doc(hidden)]
+pub mod __internal {
+    use std::{fmt::Debug, hash::Hash};
+
+    use zeroize::ZeroizeOnDrop;
+
+    use crate::CryptoKey;
+
+    /// This trait represents a key reference that can be used to identify cryptographic keys in the
+    /// key store. It is used to abstract over the different types of keys that can be used in
+    /// the system, an end user would not implement this trait directly, and would instead use
+    /// the `SymmetricKeyRef` and `AsymmetricKeyRef` traits.
+    pub trait KeyRef:
+        Debug + Clone + Copy + Hash + Eq + PartialEq + Ord + PartialOrd + Send + Sync + 'static
+    {
+        type KeyValue: CryptoKey + Send + Sync + ZeroizeOnDrop;
+
+        /// Returns whether the key is local to the current context or shared globally by the
+        /// service.
+        fn is_local(&self) -> bool;
```

Comment:
question: what does local mean?

Reply:
Oh ok I just found it 

https://github.com/bitwarden/sdk/blob/0c74d80e314f9aeaccd840fc6c9c0f31474268f2/crates/bitwarden-vault/src/cipher/cipher.rs#L175-L176